### PR TITLE
FINERACT-1022: Fixing ehcache 3

### PR DESCRIPTION
**Pull request by ptuomola**
FINERACT-1022

## Description
ehcache was not working at all after it had been upgraded from v2 to v3. This fixes the config files, dependencies and other artifacts to fix this. 

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Commit message starts with the issue number from https://issues.apache.org/jira/projects/FINERACT/. Ex: FINERACT-646 Pockets API.

- [x] Coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions have been followed.

- [x] API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm has been updated with details of any API changes.

- [x] Integration tests have been created/updated for verifying the changes made.

- [x] All Integrations tests are passing with the new commits.

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the list.)

Our guidelines for code reviews is at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide

### CODE REVIEW

**Comment by vorburger** [comment]
@xurror how does this look to you?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/ClientAddressRepository.java [hunk 1]
```diff
@@ -27,8 +27,8 @@
 public interface ClientAddressRepository extends JpaRepository<ClientAddress, Long>, JpaSpecificationExecutor<ClientAddress> {
 
     @Query("SELECT clientAddress FROM ClientAddress clientAddress WHERE clientAddress.client.id = :clientId AND clientAddress.addressType = :addressType AND clientAddress.isActive = :isActive ")
-    ClientAddress findByClientIdAndAddressTypeAndIsActive(@Param("clientId") final long clientId, final CodeValue addressType,
-            @Param("isActive") final boolean isActive);
+    ClientAddress findByClientIdAndAddressTypeAndIsActive(@Param("clientId") final long clientId,
+            @Param("addressType") final CodeValue addressType, @Param("isActive") final boolean isActive);
```
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/ClientAddressRepository.java:31]: hah, nice catch - looks like this was wrong (missing) in #1110 ?!  I missed it! @kiran-vsm FYI...

**Comment by xurror** [comment]
> @xurror how does this look to you?

I want to properly test this locally. I should be able to provide feedback in the next 24hrs

**Comment by ptuomola** [comment]
@xurror did you get a chance to try this out - did it work for you?

**Comment by vorburger** [comment]
> I want to properly test this locally. I should be able to provide feedback in the next 24hrs

@xurror any update re. this? We would love you to fully review this for FINERACT-1022. You can go ahead and merge this when you LGTM.

**Review by xurror — APPROVED** [review]
This is awesome. I tried it locally. Great work @ptuomola. Thanks again.

**Pull request merged**